src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -217,67 +240,86 @@ private ImmutableArray<Document> GetPriorityDocuments(Project project)
             return result.ToImmutable();
         }
 
-        /// <summary>
-        /// Returns <see langword="true"/> if the search found any results.
-        /// </summary>
-        private async Task<bool> ProcessProjectsAsync(
-            ImmutableArray<ImmutableArray<Project>> orderedProjects, bool isFullyLoaded, CancellationToken cancellationToken)
+        private async Task SearchOrderedProjectsAsync(
+            ImmutableArray<ImmutableArray<Project>> orderedProjects,
+            HashSet<INavigateToSearchResult> seenItems,
+            Func<INavigateToSearchService, Project, Func<INavigateToSearchResult, Task>, Task> searchProjectAsync,
+            CancellationToken cancellationToken)
         {
```

Comment:
this is the core loop that sweeps over the projects in a preferred order.  All the types of searches (normal, cached, generated-source) will defer to this loop.  All that differs is the core search they perform on each project (passed in with `searchProjectAsync`).